src/agent/rest_web_server.cpp

```diff
@@ -0,0 +1,1061 @@
+    }
+    if (mConnection->hasCallback == 0){
+    sentResponse(data,mConnection);
+    FreeConnection(fd);
+    }
+    else{
+        otbrLog(OTBR_LOG_ERR, "wait for callback");
+    }
+        
+}
+
+void RestWebServer::sentResponse(cJSON* data, Connection* connection){
+    //need more definition
+    otbrLog(OTBR_LOG_ERR, "start send");
+    char * out;
+    char * p ;
+    
+    cJSON *root,*cars;
+    root = cJSON_CreateObject();
+    cars = cJSON_CreateArray();
+    cJSON_AddItemToObject(root,"Error",cars);
+    out = cJSON_Print(root);
+    
+    if ( connection->isError == 1 ){
+    p = out;  
+    }
+    else{
+    p = cJSON_Print(data);
+    
+    }
+    char h[] = "HTTP/1.1 200 OK\r\n";
+    char h1[] = "Content-Type: application/json\r\n";
+    char h3[] = "Access-Control-Allow-Origin: *\r\n";
+    
+    write(connection->fd, h, strlen(h));
+    write(connection->fd, h1, strlen(h1));
+    write(connection->fd, h3, strlen(h3));
+    write(connection->fd, "\r\n", 2);
+    write(connection->fd,p, strlen(p));
+    
+    free(p);
+
+    // cJSON_Delete(data);
+
+}
+
+
+
+uint16_t RestWebServer::HostSwap16(uint16_t v)
+
+
+{
+    return (((v & 0x00ffU) << 8) & 0xff00) | (((v & 0xff00U) >> 8) & 0x00ff);
+}
+
+char* RestWebServer::formatBytes(const uint8_t *aBytes, uint8_t aLength){
+    char *p = new char[17]();
+    char *q = p;
+    
+    for (int i = 0; i <  aLength; i++)
+    {
+       snprintf(q,3,"%02x",aBytes[i]);
+       q+=2;
+    }
+    return p;
+}
+
+
+cJSON* RestWebServer::getJsonNodeInfo(Connection* connection,otInstance * mInstance){
+    connection->hasCallback = 0;
+    cJSON* ret = cJSON_CreateObject();
+    cJSON* cState = getJsonState(connection,mInstance);
+    cJSON* cName = getJsonNetworkName(connection,mInstance);
+    cJSON* cExtPanId = getJsonExtendedPanId(connection,mInstance);
+    cJSON* cRloc16 = getJsonRloc16(connection,mInstance);
+    cJSON* cNumRoute = getJsonNumOfRoute(connection,mInstance);
+    cJSON* cLeaderData = getJsonLeaderData(connection,mInstance);
+    cJSON* cExtAddr = getJsonExtendedAddr(connection,mInstance);
+
+    cJSON_AddItemToObject(ret, "networkName",cName);
+    cJSON_AddItemToObject(ret, "state",cState);
+    cJSON_AddItemToObject(ret, "extAddress",cExtAddr );
+    cJSON_AddItemToObject(ret, "rloc16",cRloc16 );
+    cJSON_AddItemToObject(ret, "numOfRouter",cNumRoute );
+    cJSON_AddItemToObject(ret, "leaderData",cLeaderData );
+    cJSON_AddItemToObject(ret, "extPanId",cExtPanId );
+
+    return ret;
+}
+
+cJSON*  RestWebServer::getJsonExtendedAddr(Connection* connection, otInstance * mInstance){
+    connection->hasCallback = 0;
+    cJSON* cExtAddr;
+
+    const uint8_t *extAddress = reinterpret_cast<const uint8_t *>(otLinkGetExtendedAddress(mInstance));
+    char* p = formatBytes(extAddress, OT_EXT_ADDRESS_SIZE);
+    cExtAddr  = cJSON_CreateString(p);
+
+    return cExtAddr;
+    
+ }
+
+cJSON* RestWebServer::getJsonState(Connection* connection,otInstance * mInstance){
```

Comment:
```suggestion
cJSON* RestWebServer::getJsonState(Connection* connection,otInstance * aInstance){
```

And everywhere else.